This patch fixes a bug with POST data for CGI programs

--- networking/httpd.c	2007-09-30 17:46:16.000000000 +0200
+++ networking/httpd.c	2007-10-12 14:15:01.000000000 +0200
@@ -1191,6 +1191,10 @@
 	 * What to do if CGI has closed both input and output, but
 	 * didn't exit? etc... */
 
+	if (bodyLen >= hdr_cnt)
+		bodyLen -= hdr_cnt;
+	else
+		bodyLen = 0;
 	/* NB: breaking out of this loop jumps to log_and_exit() */
 	while (1) {
 		fd_set readSet;
